app/Content/src/Http/Controllers/BlogPostController.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace App\Content\Http\Controllers;
+
+use App\Content\Http\Requests\BlogPosts\BlogPostPermissionsRequest;
+use App\Content\Http\Requests\BlogPosts\CreateBlogPostRequest;
+use App\Content\Http\Requests\BlogPosts\UpdateBlogPostRequest;
+use App\Content\Http\Resources\BlogPosts\BlogPostsResource;
+use App\Http\Controllers\Controller;
+use App\Website\Enums\BlogPostStatus;
+use App\Website\Models\Author;
+use App\Website\Models\BlogPost;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use Knuckles\Scribe\Attributes\BodyParam;
+use Knuckles\Scribe\Attributes\Endpoint;
+use Knuckles\Scribe\Attributes\Group;
+use Knuckles\Scribe\Attributes\QueryParam;
+
+class BlogPostController extends Controller
+{
+    #[Endpoint(title: 'Blog posts', description: 'This endpoint list all blog post')]
+    #[Group('Content')]
+    #[QueryParam('title', 'string', required: false, example: "?title=learnhub")]
+    #[QueryParam('tag', 'string', required: false, example: "?tag=php")]
+    #[QueryParam('author', 'string', required: false, example: "?author=john")]
+    #[QueryParam('sort', 'array', required: false, example: "?sort=[id,title]", enum: ['id', 'title', 'publish_date', 'created_at'])]
+    public function index(BlogPostPermissionsRequest $request): \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+    {
+        $blogs = BlogPost::with('author', 'tags')
+            ->when($request->has('title'), function ($query) use ($request) {
+                return $query->where('title', 'like', "%{$request->title}%");
+            })
+            ->when($request->has('tag'), function ($query) use ($request) {
+                return $query->whereHas('tags', function ($tagQuery) use ($request) {
+                    $tagQuery->where('name', 'like', "%$request->tag%");
```

Comment:
It may make sense to be able to filter by multiple tags.

Reply:
Also, it may make sense to send the full tag names instead of partially search by tag.